



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="Compare nanopore signal to find modifications">
      
      
        <link rel="canonical" href="http://nanocompore.rna.rocks/demo/SampComp_usage/">
      
      
        <meta name="author" content="Adrien Leger & Tommaso Leonardi">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.6.0">
    
    
      
        <title>Compare samples with SampComp - Nanocompore</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application.1b62728e.css">
      
      
    
    
      <script src="../../assets/javascripts/modernizr.268332fc.js"></script>
    
    
      
    
    <link rel="stylesheet" href="../../assets/fonts/material-icons.css">
    
    
      <link rel="stylesheet" href="../../css/ansi-colours.css">
    
    
      
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="http://nanocompore.rna.rocks" title="Nanocompore" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              Nanocompore
            </span>
            <span class="md-header-nav__topic">
              
                Compare samples with SampComp
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  

<a href="https://github.com/tleonardi/nanocompore/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    tleonardi/nanocompore
  </div>
</a>
          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main" role="main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="http://nanocompore.rna.rocks" title="Nanocompore" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Nanocompore
  </label>
  
    <div class="md-nav__source">
      


  

<a href="https://github.com/tleonardi/nanocompore/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    tleonardi/nanocompore
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../installation/" title="Installation instructions" class="md-nav__link">
      Installation instructions
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../data_preparation/" title="Data preparation" class="md-nav__link">
      Data preparation
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4" checked>
    
    <label class="md-nav__link" for="nav-4">
      Usage
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        Usage
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../usage/" title="How to use Nanocompore" class="md-nav__link">
      How to use Nanocompore
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../Eventalign_collapse_usage/" title="Collapse nanopolish data at kmer level" class="md-nav__link">
      Collapse nanopolish data at kmer level
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
    
    <a href="./" title="Compare samples with SampComp" class="md-nav__link md-nav__link--active">
      Compare samples with SampComp
    </a>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../SampCompDB_usage/" title="Generate plots and tables with SampCompDB" class="md-nav__link">
      Generate plots and tables with SampCompDB
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../SimReads_usage/" title="Generate simulated reads" class="md-nav__link">
      Generate simulated reads
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../alternative/" title="Alternative and complementary packages" class="md-nav__link">
      Alternative and complementary packages
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../citing/" title="Citing" class="md-nav__link">
      Citing
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../contributing/" title="Contributing" class="md-nav__link">
      Contributing
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../code_of_conduct/" title="Code of conduct" class="md-nav__link">
      Code of conduct
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../licence/" title="Licence" class="md-nav__link">
      Licence
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../changelog/" title="Changelog" class="md-nav__link">
      Changelog
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/tleonardi/nanocompore/edit/master/docs/demo/SampComp_usage.ipynb" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Compare-samples-with-SampComp">Compare samples with SampComp<a class="anchor-link" href="#Compare-samples-with-SampComp">&#182;</a></h1>
</div>
</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="In-brief">In brief<a class="anchor-link" href="#In-brief">&#182;</a></h2>
</div>
</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>First, <code>SampComp</code> parses the sample eventalign collapse files and then the observed results are piled-up per reference at position level. In a second time, positions are compared using various statistical methods and the statistics are stored in a shelve DBM database containing the results for all positions with sufficient coverage. The API returns a <code>SampCompDB</code> database wrapper object that can be subsequently interrogated to extract data and plots.</p>

</div>
</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Quick-start">Quick start<a class="anchor-link" href="#Quick-start">&#182;</a></h2>
</div>
</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Example CLI call</p>
<div class="highlight"><pre><span></span>nanocompore sampcomp <span class="se">\</span>
    --file_list1 ./data/S1_R1.tsv,./data/S1_R2.tsv <span class="se">\</span>
    --file_list2 ./data/S2_R1.tsv,./data/S2_R2.tsv <span class="se">\</span>
    --label1 S1 <span class="se">\</span>
    --label2 S2 <span class="se">\</span>
    --fasta ./reference/ref.fa <span class="se">\</span>
    --outpath ./results
</pre></div>
<p>Example API call</p>
<div class="highlight"><pre><span></span><span class="c1"># Import package</span>
<span class="kn">from</span> <span class="nn">nanocompore.SampComp</span> <span class="kn">import</span> <span class="n">SampComp</span>

<span class="c1"># Init the object</span>
<span class="n">s</span> <span class="o">=</span> <span class="n">SampComp</span><span class="p">(</span>
    <span class="n">eventalign_fn_dict</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s1">&#39;S1&#39;</span><span class="p">:{</span><span class="s1">&#39;rep1&#39;</span><span class="p">:</span><span class="s1">&#39;./data/S1_R1.tsv&#39;</span><span class="p">,</span> <span class="s1">&#39;rep2&#39;</span><span class="p">:</span><span class="s1">&#39;./data/S1_R2.tsv&#39;</span><span class="p">},</span>
        <span class="s1">&#39;S2&#39;</span><span class="p">:{</span><span class="s1">&#39;rep1&#39;</span><span class="p">:</span><span class="s1">&#39;./data/S2_R1.tsv&#39;</span><span class="p">,</span> <span class="s1">&#39;rep2&#39;</span><span class="p">:</span><span class="s1">&#39;./data/S2_R2.tsv&#39;</span><span class="p">}},</span>
    <span class="n">outpath</span> <span class="o">=</span> <span class="s2">&quot;./results/&quot;</span><span class="p">,</span>
    <span class="n">fasta_fn</span> <span class="o">=</span> <span class="s2">&quot;./reference/ref.fa&quot;</span><span class="p">)</span>

<span class="c1"># Run the analysis</span>
<span class="n">s</span><span class="p">()</span>
</pre></div>

</div>
</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Description-of-main-options">Description of main options<a class="anchor-link" href="#Description-of-main-options">&#182;</a></h2>
</div>
</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p><code>SampComp</code> provides a very flexible analysis framework with a few mandatory options and many optional parameters. The full CLI and API documentations are provided at the bottom of this page.</p>

</div>
</div>
</div>

<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="Sample-files">Sample files<a class="anchor-link" href="#Sample-files">&#182;</a></h4><p><code>SampComp</code> requires sample files obtained with <code>NanopolishComp EventalignCollapse</code> as explained before (see <a href="data_preparation.md">data preparation</a>) for both the control and the experimental conditions. 2 conditions are expected and at least 2 replicates per conditions are highly recommended. If <code>SampComp</code> is called through the CLI the files can be provided using either relevant command options or a YAML file. With the Python API you can pass either a python dictionary or a YAML file.</p>
<p>!!! info "YAML file option (CLI or API)"
    This option allows to pass a YAML formatted file indicating the sample condition labels and paths to data files with the option <code>--sample_yaml</code> for the CLI or directly with <code>eventalign_fn_dict</code> for the API. The file should be formatted as follow:</p>

<pre><code>```yaml
WT:
    rep1:   ./data/S1_R1.tsv
    rep2:   ./data/S1_R2.tsv

KO:
    rep1:   ./data/S2_R1.tsv
    rep2:   ./data/S1_R2.tsv
<div class="codehilite"><pre><span></span><span class="nt">&lt;/code&gt;&lt;/pre&gt;</span>
<span class="nt">&lt;p&gt;</span>!!! info &quot;Command line option (CLI only)&quot;
    This option requires to provide a comma separated list of files per condition using <span class="nt">&lt;code&gt;</span>--file_list1<span class="nt">&lt;/code&gt;</span> and <span class="nt">&lt;code&gt;</span>--file_list2<span class="nt">&lt;/code&gt;</span> arguments as well as the labels for condition each conditions using <span class="nt">&lt;code&gt;</span>--label1<span class="nt">&lt;/code&gt;</span> and <span class="nt">&lt;code&gt;</span>--label2<span class="nt">&lt;/code&gt;</span>.<span class="nt">&lt;/p&gt;</span>
<span class="nt">&lt;p&gt;</span>!!! info &quot;Python dictionary (API only)&quot;
    This option allows to pass a multi-level python dictionary containing the sample condition labels and paths to data files. The dictionary should be formatted as follow:<span class="nt">&lt;/p&gt;</span>

<span class="nt">&lt;pre&gt;&lt;code&gt;</span>```python
eventalign_fn_dict = {
    &quot;WT&quot;:  {&quot;rep1&quot;:&quot;./data/S1_R1.tsv&quot;, &quot;rep2&quot;:&quot;./data/S1_R2.tsv&quot;},
    &quot;KO&quot;: {&quot;rep1&quot;:&quot;./data/S2_R1.tsv&quot;, &quot;rep2&quot;:&quot;./data/S1_R2.tsv&quot;}
    }
```<span class="nt">&lt;/code&gt;&lt;/pre&gt;</span>

<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;cell border-box-sizing text_cell rendered&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;inner_cell&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;text_cell_render border-box-sizing rendered_html&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;h4</span> <span class="na">id=</span><span class="s">&quot;Transcriptome-reference-FASTA-file&quot;</span><span class="nt">&gt;</span>Transcriptome reference FASTA file<span class="nt">&lt;a</span> <span class="na">class=</span><span class="s">&quot;anchor-link&quot;</span> <span class="na">href=</span><span class="s">&quot;#Transcriptome-reference-FASTA-file&quot;</span><span class="nt">&gt;</span><span class="ni">&amp;#182;</span><span class="nt">&lt;/a&gt;&lt;/h4&gt;&lt;p&gt;</span>A transcriptome FASTA reference file is required to extract kmer sequences during the analyses. The reference has to be the same as the one used at the mapping step. (CLI: <span class="nt">&lt;code&gt;</span>--fasta<span class="nt">&lt;/code&gt;</span>, API: <span class="nt">&lt;code&gt;</span>fasta_fn<span class="nt">&lt;/code&gt;</span>)<span class="nt">&lt;/p&gt;</span>

<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;cell border-box-sizing text_cell rendered&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;inner_cell&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;text_cell_render border-box-sizing rendered_html&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;h4</span> <span class="na">id=</span><span class="s">&quot;Output-folder&quot;</span><span class="nt">&gt;</span>Output folder<span class="nt">&lt;a</span> <span class="na">class=</span><span class="s">&quot;anchor-link&quot;</span> <span class="na">href=</span><span class="s">&quot;#Output-folder&quot;</span><span class="nt">&gt;</span><span class="ni">&amp;#182;</span><span class="nt">&lt;/a&gt;&lt;/h4&gt;&lt;p&gt;</span>Although it is not mandatory, it is recommended to provide a path to a directory where the program will output the result files  (CLI: <span class="nt">&lt;code&gt;</span>--outpath<span class="nt">&lt;/code&gt;</span>, API: <span class="nt">&lt;code&gt;</span>outpath<span class="nt">&lt;/code&gt;</span>). In addition, users can also specify a prefix for the files to be generated (CLI: <span class="nt">&lt;code&gt;</span>--outprefix<span class="nt">&lt;/code&gt;</span>, API: <span class="nt">&lt;code&gt;</span>outprefix<span class="nt">&lt;/code&gt;</span>). Finally, if the outpath directory already exists, the program will raise an error to avoid erasing result files. To ignore the error you can to specify to overwrite previous results (CLI: <span class="nt">&lt;code&gt;</span>--overwrite<span class="nt">&lt;/code&gt;</span>, API: <span class="nt">&lt;code&gt;</span>overwrite<span class="nt">&lt;/code&gt;</span>).<span class="nt">&lt;/p&gt;</span>

<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;cell border-box-sizing text_cell rendered&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;inner_cell&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;text_cell_render border-box-sizing rendered_html&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;h4</span> <span class="na">id=</span><span class="s">&quot;Genome-annotation-BED-file&quot;</span><span class="nt">&gt;</span>Genome annotation BED file<span class="nt">&lt;a</span> <span class="na">class=</span><span class="s">&quot;anchor-link&quot;</span> <span class="na">href=</span><span class="s">&quot;#Genome-annotation-BED-file&quot;</span><span class="nt">&gt;</span><span class="ni">&amp;#182;</span><span class="nt">&lt;/a&gt;&lt;/h4&gt;&lt;p&gt;</span>Optionally, a BED file containing the genome annotations corresponding to the transcriptome fasta file can be provided. In that case Nanocompore will also convert the transcript coordinates into the genome space (CLI: <span class="nt">&lt;code&gt;</span>--bed<span class="nt">&lt;/code&gt;</span>, API: <span class="nt">&lt;code&gt;</span>bed_fn<span class="nt">&lt;/code&gt;</span>)<span class="nt">&lt;/p&gt;</span>

<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;cell border-box-sizing text_cell rendered&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;inner_cell&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;text_cell_render border-box-sizing rendered_html&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;h4</span> <span class="na">id=</span><span class="s">&quot;Statistical-options&quot;</span><span class="nt">&gt;</span>Statistical options<span class="nt">&lt;a</span> <span class="na">class=</span><span class="s">&quot;anchor-link&quot;</span> <span class="na">href=</span><span class="s">&quot;#Statistical-options&quot;</span><span class="nt">&gt;</span><span class="ni">&amp;#182;</span><span class="nt">&lt;/a&gt;&lt;/h4&gt;&lt;p&gt;&lt;code&gt;</span>SampComp<span class="nt">&lt;/code&gt;</span> implements several statistical methods to evaluate the difference between the 2 conditions (<span class="nt">&lt;code&gt;</span>comparison_method<span class="nt">&lt;/code&gt;</span>).<span class="nt">&lt;/p&gt;</span>
<span class="nt">&lt;ul&gt;</span>
<span class="nt">&lt;li&gt;</span>Gaussian Mixture Model = GMM (default)<span class="nt">&lt;/li&gt;</span>
<span class="nt">&lt;li&gt;</span>Kolmogorov–Smirnov test = KS (default)<span class="nt">&lt;/li&gt;</span>
<span class="nt">&lt;li&gt;</span>Mann–Whitney U test = MW<span class="nt">&lt;/li&gt;</span>
<span class="nt">&lt;li&gt;</span>T-test = TT<span class="nt">&lt;/li&gt;</span>
<span class="nt">&lt;/ul&gt;</span>
<span class="nt">&lt;p&gt;</span>In addition, it is also possible to specify the number of adjacent positions to take into account for the pvalue calculation (<span class="nt">&lt;code&gt;</span>sequence_context<span class="nt">&lt;/code&gt;</span>) as well as the weights to give to adjacent position, using either an &quot;uniform&quot; or a &quot;harmonic&quot; distribution (<span class="nt">&lt;code&gt;</span>sequence_context_weights<span class="nt">&lt;/code&gt;</span>).<span class="nt">&lt;/p&gt;</span>

<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;cell border-box-sizing text_cell rendered&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;inner_cell&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;text_cell_render border-box-sizing rendered_html&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;h4</span> <span class="na">id=</span><span class="s">&quot;Coverage-options&quot;</span><span class="nt">&gt;</span>Coverage options<span class="nt">&lt;a</span> <span class="na">class=</span><span class="s">&quot;anchor-link&quot;</span> <span class="na">href=</span><span class="s">&quot;#Coverage-options&quot;</span><span class="nt">&gt;</span><span class="ni">&amp;#182;</span><span class="nt">&lt;/a&gt;&lt;/h4&gt;&lt;p&gt;</span>The default coverage threshold for <span class="nt">&lt;code&gt;</span>SampComp<span class="nt">&lt;/code&gt;</span> to perform a statistical test is 30 reads in each replicates. This is quite conservative and can be modified if needed (<span class="nt">&lt;code&gt;</span>min_coverage<span class="nt">&lt;/code&gt;</span>). In addition, to reduce the computational burden it is possible to randomly down-sample the number of reads for high coverage references (<span class="nt">&lt;code&gt;</span>downsample_high_coverage<span class="nt">&lt;/code&gt;</span>).<span class="nt">&lt;/p&gt;</span>

<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;cell border-box-sizing text_cell rendered&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;inner_cell&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;text_cell_render border-box-sizing rendered_html&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;h4</span> <span class="na">id=</span><span class="s">&quot;Manually-exclude-or-include-references-(API-only)&quot;</span><span class="nt">&gt;</span>Manually exclude or include references (API only)<span class="nt">&lt;a</span> <span class="na">class=</span><span class="s">&quot;anchor-link&quot;</span> <span class="na">href=</span><span class="s">&quot;#Manually-exclude-or-include-references-(API-only)&quot;</span><span class="nt">&gt;</span><span class="ni">&amp;#182;</span><span class="nt">&lt;/a&gt;&lt;/h4&gt;&lt;p&gt;</span>The API allows to specify references to be included or excluded from the analysis (<span class="nt">&lt;code&gt;</span>select_ref_id<span class="nt">&lt;/code&gt;</span> and
<span class="nt">&lt;code&gt;</span>exclude_ref_id<span class="nt">&lt;/code&gt;</span>). This can be useful to analyse a specific set of transcripts or to run a small test before analysing the whole dataset.<span class="nt">&lt;/p&gt;</span>

<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;cell border-box-sizing text_cell rendered&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;inner_cell&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;text_cell_render border-box-sizing rendered_html&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;h2</span> <span class="na">id=</span><span class="s">&quot;Advanced-API-Usage&quot;</span><span class="nt">&gt;</span>Advanced API Usage<span class="nt">&lt;a</span> <span class="na">class=</span><span class="s">&quot;anchor-link&quot;</span> <span class="na">href=</span><span class="s">&quot;#Advanced-API-Usage&quot;</span><span class="nt">&gt;</span><span class="ni">&amp;#182;</span><span class="nt">&lt;/a&gt;&lt;/h2&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;cell border-box-sizing text_cell rendered&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;inner_cell&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;text_cell_render border-box-sizing rendered_html&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;h3</span> <span class="na">id=</span><span class="s">&quot;Import-the-package&quot;</span><span class="nt">&gt;</span>Import the package<span class="nt">&lt;a</span> <span class="na">class=</span><span class="s">&quot;anchor-link&quot;</span> <span class="na">href=</span><span class="s">&quot;#Import-the-package&quot;</span><span class="nt">&gt;</span><span class="ni">&amp;#182;</span><span class="nt">&lt;/a&gt;&lt;/h3&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;/div&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;cell border-box-sizing code_cell rendered&quot;</span><span class="nt">&gt;</span>
<span class="nt">&lt;div</span> <span class="na">class=</span><span class="s">&quot;input&quot;</span><span class="nt">&gt;</span>

```python
from nanocompore.SampComp import SampComp
from nanocompore.common import *
import shutil

# Create output dir and define log level
outdir = &quot;results&quot;
shutil.rmtree (outdir)
mkdir(outdir, exist_ok=True)
set_logger(&quot;info&quot;)
</pre></div>

</div>

</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Using-a-Python-dictionary-to-specify-the-locations-of-the-eventalign-files">Using a Python dictionary to specify the locations of the eventalign files<a class="anchor-link" href="#Using-a-Python-dictionary-to-specify-the-locations-of-the-eventalign-files">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Default option using a multilevel dictionary.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="codehilite"><pre><span></span><span class="c1"># Init the object</span>
<span class="n">s</span> <span class="o">=</span> <span class="n">SampComp</span> <span class="p">(</span> 
    <span class="n">eventalign_fn_dict</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s2">&quot;KO&quot;</span><span class="p">:{</span>
            <span class="s2">&quot;rep1&quot;</span><span class="p">:</span><span class="s1">&#39;./eventalign_files/yeast/KO_1_eventalign_collapsed.tsv&#39;</span><span class="p">,</span> 
            <span class="s2">&quot;rep2&quot;</span><span class="p">:</span><span class="s2">&quot;./eventalign_files/yeast/KO_2_eventalign_collapsed.tsv&quot;</span><span class="p">},</span>
        <span class="s2">&quot;WT&quot;</span><span class="p">:{</span>
            <span class="s2">&quot;rep1&quot;</span><span class="p">:</span><span class="s1">&#39;./eventalign_files/yeast/WT_1_eventalign_collapsed.tsv&#39;</span><span class="p">,</span> 
            <span class="s2">&quot;rep2&quot;</span><span class="p">:</span><span class="s2">&quot;./eventalign_files/yeast/WT_2_eventalign_collapsed.tsv&quot;</span><span class="p">}},</span>
    <span class="n">outpath</span> <span class="o">=</span> <span class="s2">&quot;./results&quot;</span><span class="p">,</span>
    <span class="n">outprefix</span><span class="o">=</span> <span class="s2">&quot;yeast_&quot;</span><span class="p">,</span>
    <span class="n">nthreads</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span>
    <span class="n">max_invalid_kmers_freq</span> <span class="o">=</span> <span class="mf">0.2</span><span class="p">,</span>
    <span class="n">min_coverage</span><span class="o">=</span><span class="mi">30</span><span class="p">,</span>
    <span class="n">fasta_fn</span> <span class="o">=</span> <span class="s2">&quot;./references/yeast/Yeast_transcriptome.fa&quot;</span><span class="p">,</span>
    <span class="n">overwrite</span> <span class="o">=</span> <span class="kc">True</span><span class="p">)</span>

<span class="c1"># Run the analysis</span>
<span class="n">db</span> <span class="o">=</span> <span class="n">s</span><span class="p">()</span>
</pre></div>

</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">
<div class="output_subarea output_stream output_stderr output_text">
<pre>2020-12-07T18:30:16.356391+0000 INFO - MainProcess | Checking and initialising SampComp
2020-12-07T18:30:16.419459+0000 INFO - MainProcess | Reading eventalign index files
2020-12-07T18:30:16.431004+0000 ERROR - MainProcess | High fraction of invalid kmers (20.74%) for read 107
2020-12-07T18:30:16.432924+0000 INFO - MainProcess |    References found in index: 146
2020-12-07T18:30:16.433256+0000 INFO - MainProcess | Filtering out references with low coverage
2020-12-07T18:30:16.613162+0000 INFO - MainProcess |    References remaining after reference coverage filtering: 1
2020-12-07T18:30:16.613684+0000 INFO - MainProcess | Starting data processing
2020-12-07T18:31:08.593363+0000 INFO - Process-17 | All Done. Transcripts processed: 1
2020-12-07T18:31:08.636924+0000 INFO - MainProcess | Loading SampCompDB
2020-12-07T18:31:08.654674+0000 INFO - MainProcess | Calculate results
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Using-a-YAML-file-instead-to-specify-the-files-location">Using a YAML file instead to specify the files location<a class="anchor-link" href="#Using-a-YAML-file-instead-to-specify-the-files-location">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>To use a YAML file, just pass a path to the file instead of a dictionary. In this example we also specify to use 10 threads to speed-up the data processing (<code>nthreads = 10</code>).</p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="codehilite"><pre><span></span><span class="c1"># Read YAML file to display structure</span>
<span class="nb">print</span><span class="p">(</span><span class="nb">open</span><span class="p">(</span><span class="s2">&quot;./samples.yaml&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">read</span><span class="p">())</span>
</pre></div>

</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">
<div class="output_subarea output_stream output_stdout output_text">
<pre>Modified:
    rep1:   ./eventalign_files/simulated/modified_rep_1.tsv
    rep2:   ./eventalign_files/simulated/modified_rep_2.tsv
Unmodified:
    rep1:   ./eventalign_files/simulated/unmodified_rep_1.tsv
    rep2:   ./eventalign_files/simulated/unmodified_rep_2.tsv

</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="codehilite"><pre><span></span><span class="c1"># Init the object</span>
<span class="n">s</span> <span class="o">=</span> <span class="n">SampComp</span> <span class="p">(</span>
    <span class="n">eventalign_fn_dict</span> <span class="o">=</span> <span class="s2">&quot;./samples.yaml&quot;</span><span class="p">,</span>
    <span class="n">outpath</span><span class="o">=</span> <span class="s2">&quot;./results&quot;</span><span class="p">,</span>
    <span class="n">outprefix</span><span class="o">=</span> <span class="s2">&quot;simulated_&quot;</span><span class="p">,</span>
    <span class="n">fasta_fn</span> <span class="o">=</span> <span class="s2">&quot;./references/simulated/ref.fa&quot;</span><span class="p">,</span>
    <span class="n">overwrite</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
    <span class="n">progress</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
    <span class="n">nthreads</span> <span class="o">=</span> <span class="mi">4</span><span class="p">)</span>

<span class="c1"># Run the analysis</span>
<span class="n">db</span> <span class="o">=</span> <span class="n">s</span> <span class="p">()</span>
</pre></div>

</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">
<div class="output_subarea output_stream output_stderr output_text">
<pre>2020-12-07T18:31:19.526366+0000 INFO - MainProcess | Checking and initialising SampComp
2020-12-07T18:31:19.536530+0000 INFO - MainProcess | Reading eventalign index files
2020-12-07T18:31:19.559451+0000 INFO - MainProcess |    References found in index: 5
2020-12-07T18:31:19.559797+0000 INFO - MainProcess | Filtering out references with low coverage
2020-12-07T18:31:19.563239+0000 INFO - MainProcess |    References remaining after reference coverage filtering: 5
2020-12-07T18:31:19.566024+0000 INFO - MainProcess | Starting data processing
100%|██████████| 5/5 [00:13&lt;00:00,  2.63s/ Processed References]
2020-12-07T18:31:32.819370+0000 INFO - Process-21 | All Done. Transcripts processed: 5
2020-12-07T18:31:32.828972+0000 INFO - MainProcess | Loading SampCompDB
2020-12-07T18:31:32.841026+0000 INFO - MainProcess | Calculate results
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Tweaking-the-statistical-tests">Tweaking the statistical tests<a class="anchor-link" href="#Tweaking-the-statistical-tests">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>In this example, we require all 4 possible statistical tests to be performed (<code>comparison_methods=["GMM", "MW", "KS", "TT"]</code>), and we want a pvalue that takes into account 2 adjacent positions(<code>sequence_context=2</code>). In addition we also specify that the weights for adjacent position should follow a harmonic distribution (<code>sequence_context_weights="harmonic"</code>). Finally, we also want the Gaussian mixture model to use the logistic regression test on top of the ANOVA (<code>logit = True</code>).</p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="codehilite"><pre><span></span><span class="c1"># Init the object</span>
<span class="n">s</span> <span class="o">=</span> <span class="n">SampComp</span> <span class="p">(</span>
    <span class="n">eventalign_fn_dict</span> <span class="o">=</span> <span class="s2">&quot;./samples.yaml&quot;</span><span class="p">,</span>
    <span class="n">outpath</span> <span class="o">=</span> <span class="s2">&quot;./results/&quot;</span><span class="p">,</span>
    <span class="n">outprefix</span><span class="o">=</span> <span class="s2">&quot;simulated_stats_&quot;</span><span class="p">,</span>
    <span class="n">fasta_fn</span> <span class="o">=</span> <span class="s2">&quot;./references/simulated/ref.fa&quot;</span><span class="p">,</span>
    <span class="n">overwrite</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
    <span class="n">comparison_methods</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;GMM&quot;</span><span class="p">,</span> <span class="s2">&quot;MW&quot;</span><span class="p">,</span> <span class="s2">&quot;KS&quot;</span><span class="p">,</span> <span class="s2">&quot;TT&quot;</span><span class="p">],</span>
    <span class="n">sequence_context</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span>
    <span class="n">sequence_context_weights</span><span class="o">=</span><span class="s2">&quot;harmonic&quot;</span><span class="p">,</span>
    <span class="n">logit</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
    <span class="n">nthreads</span> <span class="o">=</span> <span class="mi">6</span><span class="p">)</span>

<span class="c1"># Run the analysis</span>
<span class="n">db</span> <span class="o">=</span> <span class="n">s</span> <span class="p">()</span>
</pre></div>

</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">
<div class="output_subarea output_stream output_stderr output_text">
<pre>2020-12-07T18:31:48.171748+0000 INFO - MainProcess | Checking and initialising SampComp
2020-12-07T18:31:48.176382+0000 INFO - MainProcess | Reading eventalign index files
2020-12-07T18:31:48.189340+0000 INFO - MainProcess |    References found in index: 5
2020-12-07T18:31:48.189937+0000 INFO - MainProcess | Filtering out references with low coverage
2020-12-07T18:31:48.193885+0000 INFO - MainProcess |    References remaining after reference coverage filtering: 5
2020-12-07T18:31:48.195954+0000 INFO - MainProcess | Starting data processing
2020-12-07T18:32:08.969484+0000 INFO - Process-27 | All Done. Transcripts processed: 5
2020-12-07T18:32:08.983224+0000 INFO - MainProcess | Loading SampCompDB
2020-12-07T18:32:09.000121+0000 INFO - MainProcess | Calculate results
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Explore-the-database-object-returned-by-Sampcomp">Explore the database object returned by Sampcomp<a class="anchor-link" href="#Explore-the-database-object-returned-by-Sampcomp">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>The database object returned by <code>Sampcomp</code> is a Python GDBM object database indexed by reference id. Although we highly recommend to used <code>SampCompDB</code> to extract information, it is also possible to open the database directly using the python standard library module <a href="https://docs.python.org/3.6/library/shelve.html">shelve</a>.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="codehilite"><pre><span></span><span class="kn">import</span> <span class="nn">shelve</span>

<span class="k">with</span> <span class="n">shelve</span><span class="o">.</span><span class="n">open</span> <span class="p">(</span><span class="s2">&quot;./results/simulated_SampComp.db&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">db</span><span class="p">:</span>
    <span class="c1"># Read metadata stored in metadata slot</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">db</span><span class="p">[</span><span class="s2">&quot;__metadata&quot;</span><span class="p">])</span>
    <span class="c1"># Read list of references ids stored in ref_id_list slot</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">db</span><span class="p">[</span><span class="s2">&quot;__ref_id_list&quot;</span><span class="p">])</span>
    <span class="c1"># Access stats for reference &#39;ref_0000&#39; in position 1</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">db</span><span class="p">[</span><span class="s2">&quot;ref_0000&quot;</span><span class="p">][</span><span class="mi">1</span><span class="p">][</span><span class="s2">&quot;txComp&quot;</span><span class="p">])</span>
</pre></div>

</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">
<div class="output_subarea output_stream output_stdout output_text">
<pre>{&#39;package_name&#39;: &#39;1.0.1.dev0&#39;, &#39;package_version&#39;: &#39;nanocompore&#39;, &#39;timestamp&#39;: &#39;2020-12-07 18:31:32.813597&#39;, &#39;comparison_methods&#39;: [&#39;GMM&#39;, &#39;KS&#39;], &#39;pvalue_tests&#39;: [&#39;GMM_anova_pvalue&#39;, &#39;KS_dwell_pvalue&#39;, &#39;KS_intensity_pvalue&#39;], &#39;sequence_context&#39;: 0, &#39;min_coverage&#39;: 30, &#39;n_samples&#39;: 4}
[&#39;ref_0001&#39;, &#39;ref_0000&#39;, &#39;ref_0002&#39;, &#39;ref_0003&#39;, &#39;ref_0004&#39;]
{&#39;GMM_model&#39;: {&#39;model&#39;: GaussianMixture(n_components=2,
                random_state=RandomState(MT19937) at 0x7FE80C820270), &#39;cluster_counts&#39;: &#39;Modified_rep1:43/12__Modified_rep2:49/11__Unmodified_rep1:4/51__Unmodified_rep2:3/57&#39;}, &#39;GMM_anova_pvalue&#39;: 0.0026070675453166692, &#39;GMM_anova_model&#39;: {&#39;pvalue&#39;: 0.0026070675453166692, &#39;delta_logit&#39;: 3.8311555435, &#39;table&#39;: F_onewayResult(statistic=382.0733766141018, pvalue=0.0026070675453166692), &#39;log_ratios&#39;: array([ 1.21924028,  1.42711636, -2.34180581, -2.67414865])}, &#39;KS_intensity_pvalue&#39;: 2.0829525794874904e-18, &#39;KS_dwell_pvalue&#39;: 4.9251621679434625e-36, &#39;shift_stats&#39;: OrderedDict([(&#39;c1_mean_intensity&#39;, 99.54709799468591), (&#39;c2_mean_intensity&#39;, 91.84415815799566), (&#39;c1_median_intensity&#39;, 100.61305720731731), (&#39;c2_median_intensity&#39;, 92.05605629218894), (&#39;c1_sd_intensity&#39;, 6.85551651760893), (&#39;c2_sd_intensity&#39;, 5.143626863495356), (&#39;c1_mean_dwell&#39;, 0.02567361308475615), (&#39;c2_mean_dwell&#39;, 0.009745078440682191), (&#39;c1_median_dwell&#39;, 0.023047132241771167), (&#39;c2_median_dwell&#39;, 0.007591605413309661), (&#39;c1_sd_dwell&#39;, 0.010681688969392145), (&#39;c2_sd_dwell&#39;, 0.007203231747770627)])}
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Full-CLI-and-API-documentations">Full CLI and API documentations<a class="anchor-link" href="#Full-CLI-and-API-documentations">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="API-documentation">API documentation<a class="anchor-link" href="#API-documentation">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>API help can be obtained with conventional python methods (help or ?) or rendered nicely in Jupyter with the jhelp function from nanocompore</p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="codehilite"><pre><span></span><span class="kn">from</span> <span class="nn">nanocompore.SampComp</span> <span class="kn">import</span> <span class="n">SampComp</span>
<span class="kn">from</span> <span class="nn">nanocompore.common</span> <span class="kn">import</span> <span class="n">jhelp</span>
<span class="n">jhelp</span><span class="p">(</span><span class="n">SampComp</span><span class="p">)</span>
</pre></div>

</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

<div class="output_markdown rendered_html output_subarea ">
<p><strong>SampComp</strong> (eventalign_fn_dict, fasta_fn, bed_fn, outpath, outprefix, overwrite, whitelist, comparison_methods, logit, allow_warnings, sequence_context, sequence_context_weights, min_coverage, min_ref_length, downsample_high_coverage, max_invalid_kmers_freq, select_ref_id, exclude_ref_id, nthreads, progress)</p>
<p>Init analysis and check args</p>
<hr>
<ul>
<li><strong>eventalign_fn_dict</strong> (required) [dict]</li>
</ul>
<p>Multilevel dictionnary indicating the condition_label, sample_label and file name of the eventalign_collapse output. 2 conditions are expected and at least 2 sample replicates per condition are highly recommended. One can also pass YAML file describing the samples instead. Example <code>d = {"S1": {"R1":"path1.tsv", "R2":"path2.tsv"}, "S2": {"R1":"path3.tsv", "R2":"path4.tsv"}}</code></p>
<ul>
<li><strong>fasta_fn</strong> (required) [str]</li>
</ul>
<p>Path to a fasta file corresponding to the reference used for read alignment.</p>
<ul>
<li><strong>bed_fn</strong> (default: None) [str]</li>
</ul>
<p>Path to a BED file containing the annotation of the transcriptome used as reference when mapping.</p>
<ul>
<li><strong>outpath</strong> (default: results) [str]</li>
</ul>
<p>Path to the output folder.</p>
<ul>
<li><strong>outprefix</strong> (default: out_) [str]</li>
</ul>
<p>text outprefix for all the files generated by the function.</p>
<ul>
<li><strong>overwrite</strong> (default: False) [bool]</li>
</ul>
<p>If the output directory already exists, the standard behaviour is to raise an error to prevent overwriting existing data This option ignore the error and overwrite data if they have the same outpath and outprefix.</p>
<ul>
<li><strong>whitelist</strong> (default: None) [Whitelist]</li>
</ul>
<p>Whitelist object previously generated with nanocompore Whitelist. If not given, will be automatically generated.</p>
<ul>
<li><strong>comparison_methods</strong> (default: ['GMM', 'KS']) [list]</li>
</ul>
<p>Statistical method to compare the 2 samples (mann_whitney or MW, kolmogorov_smirnov or KS, t_test or TT, gaussian_mixture_model or GMM). This can be a list or a comma separated string. {MW,KS,TT,GMM}</p>
<ul>
<li><strong>logit</strong> (default: False) [bool]</li>
</ul>
<p>Force logistic regression even if we have less than 2 replicates in any condition.</p>
<ul>
<li><strong>allow_warnings</strong> (default: False) [bool]</li>
</ul>
<p>If True runtime warnings during the ANOVA tests don't raise an error.</p>
<ul>
<li><strong>sequence_context</strong> (default: 0) [int]</li>
</ul>
<p>Extend statistical analysis to contigous adjacent base if available.</p>
<ul>
<li><strong>sequence_context_weights</strong> (default: uniform) [str]</li>
</ul>
<p>type of weights to used for combining p-values. {uniform,harmonic}</p>
<ul>
<li><strong>min_coverage</strong> (default: 30) [int]</li>
</ul>
<p>minimal read coverage required in all sample.</p>
<ul>
<li><strong>min_ref_length</strong> (default: 100) [int]</li>
</ul>
<p>minimal length of a reference transcript to be considered in the analysis</p>
<ul>
<li><strong>downsample_high_coverage</strong> (default: 1000) [int]</li>
</ul>
<p>For reference with higher coverage, downsample by randomly selecting reads.</p>
<ul>
<li><strong>max_invalid_kmers_freq</strong> (default: 0.1) [float]</li>
</ul>
<p>maximum frequency of NNNNN, mismatching and missing kmers in reads.</p>
<ul>
<li><strong>select_ref_id</strong> (default: []) [list]</li>
</ul>
<p>if given, only reference ids in the list will be selected for the analysis.</p>
<ul>
<li><strong>exclude_ref_id</strong> (default: []) [list]</li>
</ul>
<p>if given, refid in the list will be excluded from the analysis.</p>
<ul>
<li><strong>nthreads</strong> (default: 3) [int]</li>
</ul>
<p>Number of threads (two are used for reading and writing, all the others for parallel processing).</p>
<ul>
<li><strong>progress</strong> (default: False) [bool]</li>
</ul>
<p>Display a progress bar during execution</p>

</div>

</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="CLI-documentation">CLI documentation<a class="anchor-link" href="#CLI-documentation">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<div class="highlight"><pre><span></span>nanocompore sampcomp --help
</pre></div>
<div class="highlight"><pre><span></span>usage: nanocompore sampcomp [-h] [--sample_yaml sample_yaml]
                            [--file_list1 /path/to/Condition1_rep1,/path/to/Condition1_rep2]
                            [--file_list2 /path/to/Condition2_rep1,/path/to/Condition2_rep2]
                            [--label1 Condition1] [--label2 Condition2]
                            --fasta FASTA [--bed BED]
                            [--max_invalid_kmers_freq MAX_INVALID_KMERS_FREQ]
                            [--min_coverage MIN_COVERAGE]
                            [--downsample_high_coverage DOWNSAMPLE_HIGH_COVERAGE]
                            [--min_ref_length MIN_REF_LENGTH]
                            [--comparison_methods COMPARISON_METHODS]
                            [--sequence_context {0,1,2,3,4}]
                            [--sequence_context_weights {uniform,harmonic}]
                            [--pvalue_thr PVALUE_THR] [--logit]
                            [--allow_warnings] [--nthreads NTHREADS]
                            [--outpath OUTPATH] [--outprefix OUTPREFIX]
                            [--overwrite] [--log_level {warning,info,debug}]
                            [--progress]

Compare 2 samples and find significant signal differences

* Minimal example with file_list arguments

    nanocompore sampcomp -1 f1.tsv,f2.tsv -2 f3.tsv,f4.tsv -f ref.fa -o results
* Minimal example with sample YAML file

    nanocompore sampcomp -y samples.yaml -f ref -o results

optional arguments:
  -h, --help            show this help message and exit

YAML sample files:
  Option allowing to describe sample files in a YAML file

  --sample_yaml sample_yaml, -y sample_yaml
                        YAML file containing the sample file labels. See
                        formatting in documentation. (required if --file_list1
                        and --file_list2 not given)

Arguments sample files:
  Option allowing to describe sample files directly as command line arguments

  --file_list1 /path/to/Condition1_rep1,/path/to/Condition1_rep2, -1 /path/to/Condition1_rep1,/path/to/Condition1_rep2
                        Comma separated list of NanopolishComp files for label
                        1. (required if --sample_yaml not given)
  --file_list2 /path/to/Condition2_rep1,/path/to/Condition2_rep2, -2 /path/to/Condition2_rep1,/path/to/Condition2_rep2
                        Comma separated list of NanopolishComp files for label
                        2. (required if --sample_yaml not given)
  --label1 Condition1   Label for files in --file_list1 (default: Condition1)
  --label2 Condition2   Label for files in --file_list2 (default: Condition2)

Input options:
  --fasta FASTA, -f FASTA
                        Fasta file used for mapping (required)
  --bed BED             BED file with annotation of transcriptome used for
                        mapping (optional)

Transcript filtering options:
  --max_invalid_kmers_freq MAX_INVALID_KMERS_FREQ
                        Max fequency of invalid kmers (default: 0.1)
  --min_coverage MIN_COVERAGE
                        Minimum coverage required in each condition to do the
                        comparison (default: 30)
  --downsample_high_coverage DOWNSAMPLE_HIGH_COVERAGE
                        Transcripts with high coverage will be downsampled
                        (default: 5000)
  --min_ref_length MIN_REF_LENGTH
                        Minimum length of a reference transcript to include it
                        in the analysis (default: 100)

Statistical testing options:
  --comparison_methods COMPARISON_METHODS
                        Comma separated list of comparison methods. Valid
                        methods are: GMM,KS,TT,MW. (default: GMM,KS)
  --sequence_context {0,1,2,3,4}
                        Sequence context for combining p-values (default: 0)
  --sequence_context_weights {uniform,harmonic}
                        Type of weights to use for combining p-values
  --pvalue_thr PVALUE_THR
                        Adjusted p-value threshold for reporting significant
                        sites (default: 0.05)
  --logit               Use logistic regression testing also when all
                        conditions have replicates (default: False)
  --allow_warnings      If True runtime warnings during the ANOVA tests don&#39;t
                        raise an error (default: False)

Other options:
  --nthreads NTHREADS, -t NTHREADS
                        Number of threads (default: 3)

Output options:
  --outpath OUTPATH, -o OUTPATH
                        Path to the output folder (default: ./)
  --outprefix OUTPREFIX, -p OUTPREFIX
                        text outprefix for all the files generated (default:
                        out)
  --overwrite, -w       Use --outpath even if it exists already (default:
                        False)

Verbosity options:
  --log_level {warning,info,debug}
                        Set the log level (default: info)
  --progress            Display a progress bar during execution (default:
                        False)
</pre></div>

</div>
</div>
</div>
                
                  
                
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../Eventalign_collapse_usage/" title="Collapse nanopolish data at kmer level" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Collapse nanopolish data at kmer level
              </span>
            </div>
          </a>
        
        
          <a href="../SampCompDB_usage/" title="Generate plots and tables with SampCompDB" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Generate plots and tables with SampCompDB
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2019 Adrien Leger & Tommaso Leonardi
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application.808e90bb.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:"../.."}})</script>
      
    
  </body>
</html>