# How to use Nanocompore

Nanocompore was designed to be used either through a python API or a command line interface. The package contains 3 main modules: `SampComp`, `SampCompDB` and `SimReads`

### Eventalign_collapse

As its name suggests `Eventalign_collapse`, collapses nanopolish eventalign files at kmer level. This step drastically reduces the size of the resquiggled file and allow to index the data by read and reference, which allows fast random access to the file by `SampComp`. This step is required and described in the data preparation section. It can be run either through the API (`nanocompore.Eventalign_collapse.Eventalign_collapse`) or the CLI (`nanocompore eventalign_collapse`).

* [SampComp Usage](https://nanocompore.rna.rocks/demo/Eventalign_collapse_usage/)

### SampComp

This is the main module which compares the signal of 2 experimental conditions. It takes Nanopolished datasets as input and generates a database containing all the results as output. `SampComp` has to be run first either through the API (`nanocompore.SampComp.SampComp`) or the CLI (`nanocompore sampcomp`).

* [SampComp Usage](https://nanocompore.rna.rocks/demo/SampComp_usage/)


### SampCompDB

`SampCompDB` is a wrapper around the DBM object database generated by `SampComp`. This module performs secondary statistical analyses and provide simple high level functions to plot, explore and export the results. At the moment `SampCompDB` is only accessible through the interactive python API (`nanocompore.SampCompDB.SampCompDB`). We strongly recommend to use [jupyter notebook](https://jupyter.org/).

* [SampCompDB Usage](https://nanocompore.rna.rocks/demo/SampCompDB_usage/)

### SimReads

This module can be used to generate artificial datasets based on a model file obtained from IVT generated RNA sequenced by direct RNA sequencing ([Datasets](https://github.com/nanopore-wgs-consortium/NA12878/blob/master/nanopore-human-transcriptome/fastq_fast5_bulk.md), from the Nanopore RNA consortium). In addition, one can also simulate the presence of modifications by allowing to deviate from the model for selected positions.

* [Simulate_reads Usage](https://nanocompore.rna.rocks/demo/SimReads_usage)
